azurerm/helpers/azure/app_service.go

```diff
@@ -332,6 +332,54 @@ func SchemaAppServiceSiteConfig() *schema.Schema {
 					},
 				},
 
+				"scm_use_main_ip_restriction": {
+					Type:     schema.TypeBool,
+					Optional: true,
+					Default:  false,
+				},
+
+				"scm_ip_restriction": {
+					Type:       schema.TypeList,
+					Optional:   true,
+					Computed:   true,
+					ConfigMode: schema.SchemaConfigModeAttr,
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"ip_address": {
+								Type:         schema.TypeString,
+								Optional:     true,
+								ValidateFunc: validate.CIDR,
+							},
+							"virtual_network_subnet_id": {
+								Type:         schema.TypeString,
+								Optional:     true,
+								ValidateFunc: validation.StringIsNotEmpty,
+							},
+							"name": {
+								Type:         schema.TypeString,
+								Optional:     true,
+								Computed:     true,
+								ValidateFunc: validation.StringIsNotEmpty,
+							},
+							"priority": {
+								Type:         schema.TypeInt,
+								Optional:     true,
+								Computed:     true,
+								ValidateFunc: validation.IntBetween(1, 2147483647),
+							},
```

Comment:
This should probably not be computed, but have a default of `65000`.  If any restriction is given a priority, then all must have one.
```suggestion
							"priority": {
								Type:         schema.TypeInt,
								Optional:     true,
								Default:      65000,
								ValidateFunc: validation.IntBetween(1, 2147483647),
							},
```
This is also the case for the "main" `ip_restriction`, that can be addressed separately (it's on my todo list...) to keep the scope of this PR focused.